src/methods/Method_E.cpp

```diff
@@ -0,0 +1,257 @@
+#include "Method_E.h"
+
+#include "../Analysis.h"
+#include "../general/Logger.h"
+
+#include <fstream>
+#include <iostream>
+
+using std::ofstream;
+using std::string;
+using std::vector;
+
+Method_E::Method_E()
+{
+    _minFrame        = NULL;
+    _deltaT          = 100;
+    _fps             = 16;
+    _lineForMethod_E = nullptr;
+    _boxForMethod_E  = nullptr;
+    _fRho            = nullptr;
+    _fFlow           = nullptr;
+    _lenLine         = NULL;
+    _firstFrame      = nullptr;
+}
+
+Method_E::~Method_E() {}
+
+bool Method_E::Process(
+    const PedData & peddata,
+    const fs::path & scriptsLocation,
+    const double & zPos_measureArea)
+{
+    _trajName        = peddata.GetTrajName();
+    _projectRootDir  = peddata.GetProjectRootDir();
+    _scriptsLocation = scriptsLocation;
+    _outputLocation  = peddata.GetOutputLocation();
+    _peds_t          = peddata.GetPedIDsByFrameNr();
+    _xCor            = peddata.GetXCor();
+    _yCor            = peddata.GetYCor();
+    _minFrame        = peddata.GetMinFrame();
+    _fps             = peddata.GetFps();
+    _firstFrame = peddata.GetFirstFrame();
+
+    if(_lineForMethod_E) {
+        HandleLineMeasurementArea(peddata, scriptsLocation, zPos_measureArea);
+    } else if(_boxForMethod_E) {
+        HandleBoxMeasurementArea(peddata, scriptsLocation, zPos_measureArea);
+    }
+    
+    return true;
+}
+
+void Method_E::HandleLineMeasurementArea(
+    const PedData & peddata,
+    const fs::path & scriptsLocation,
+    const double & zPos_measureArea)
+{
+    _measureAreaId = boost::lexical_cast<string>(_lineForMethod_E->_id);
+
+    for(int i = 0; i < peddata.GetNumPeds(); i++) {
+        _passLine.push_back(false);
+    }
+
+    _lenLine = sqrt(
+                   pow((_lineForMethod_E->_lineStartX - _lineForMethod_E->_lineEndX), 2) +
+                   pow((_lineForMethod_E->_lineStartY - _lineForMethod_E->_lineEndY), 2)) *
+               CMtoM;
+    // length of line -> delta x for density
+
+    OpenRhoFileMethodE();
+    LOG_INFO("------------------------Analyzing with Method E-----------------------------");
+    LOG_INFO("Line is selected as measurement area (id {})", _measureAreaId);
+
+    for(const auto & [frameNr, ids] : _peds_t) {
+        int frid = frameNr + _minFrame;
+
+        if(!(frid % 100)) {
```

Comment:
Clang-Tidy: Implicit conversion 'int' -> bool.

```suggestion
        if((frid % 100) == 0) {
```